tests/infra.py

```diff
@@ -132,39 +132,29 @@ def __getattr__(self, item: str) -> typing.Any:
             raise AttributeError(item)
 
 
-class S3TestBundle:
-    """
-    A test bundle staged in S3
-
-    This class does a little bit of "double duty" as we also use it to store the uuid and versions used with the API
-    """
-    BUCKET_TEST_FIXTURES = get_env('DSS_S3_BUCKET_TEST_FIXTURES')
-
-    def __init__(self, path, bucket=BUCKET_TEST_FIXTURES):
-        self.bucket = boto3.resource('s3').Bucket(bucket)
+class TestBundle:
+    def __init__(self, blobstore: BlobStore, path: str, bucket: str) -> None:
         self.path = path
-        self.files = self.enumerate_bundle_files()
         self.uuid = str(uuid.uuid4())
         self.version = None
+        self.blobstore = blobstore
+        self.bucket = bucket
+        self.files = self.enumerate_bundle_files()
 
-    def enumerate_bundle_files(self):
-        object_summaries = self.bucket.objects.filter(Prefix=f"{self.path}/")
-        return [S3File(objectSummary, self) for objectSummary in object_summaries]
-
+    def enumerate_bundle_files(self) -> list:
+        object_summaries = self.blobstore.list(self.bucket, prefix=f"{self.path}/")
+        return [TestFile(object_summary, self) for object_summary in object_summaries]
 
-class S3File:
-    """
-    A test file staged in S3
-    """
 
-    def __init__(self, object_summary, bundle):
+class TestFile:
+    def __init__(self, object_summary, bundle) -> None:
         self.bundle = bundle
-        self.path = object_summary.key
-        self.metadata = object_summary.Object().metadata
+        self.metadata = bundle.blobstore.get_user_metadata(bundle.bucket, object_summary)
         self.indexed = True if self.metadata['hca-dss-content-type'] == "application/json" else False
-        self.name = os.path.basename(self.path)
-        self.url = f"s3://{bundle.bucket.name}/{self.path}"
+        self.name = os.path.basename(object_summary)
+        self.path = object_summary
         self.uuid = str(uuid.uuid4())
+        self.url = f"s3://{bundle.bucket}/{self.path}"
```

Comment:
If `TestFile` is not to be cloud-specific, then this is not the right source url.

Reply:
You're absolutely right.